disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/FromDiskExporterBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.contrib.disk.buffering.internal.StorageConfiguration;
+import io.opentelemetry.contrib.disk.buffering.internal.serialization.serializers.SignalSerializer;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.Storage;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.StorageBuilder;
+import io.opentelemetry.sdk.common.Clock;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
+
+public class FromDiskExporterBuilder<T> {
+
+  private SignalSerializer<T> serializer = noopSerializer();
+  private Function<Collection<T>, CompletableResultCode> exportFunction =
+      x -> CompletableResultCode.ofFailure();
+
+  @NotNull
+  private static <T> SignalSerializer<T> noopSerializer() {
+    return new SignalSerializer<T>() {
+
+      @Override
+      public byte[] serialize(Collection<T> ts) {
+        return new byte[0];
+      }
+
+      @Override
+      public List<T> deserialize(byte[] source) {
+        return Collections.emptyList();
+      }
+    };
+  }
+
+  private final StorageBuilder storageBuilder = Storage.builder();
+
+  @CanIgnoreReturnValue
+  public FromDiskExporterBuilder<T> setFolderName(String folderName) {
```

Comment:
The folder name and the serializer must be the same as the ones used in each `ToDiskExporter` impl for it to work properly, so I'm a bit concerned about providing this builder publicly as it seems like it could be easy to misconfigure it. Instead, I would move it (and also `FromDiskExporter.java`) to the internal package and create public implementations for each signal in this package, similarly to what's done for the `ToDiskExporter`s.

Reply:
Ok, I gave that a shot....and it ended up working out and now there's nice parity between the `ToDisk` and the `FromDisk` per signal types. Thanks for the idea!

Unfortunately, it ended up being a LOT of work to get the tests shored up. For starters, we were several versions behind on the protobufs...and I got us on the latest. The latest, tho, includes the `TraceFlags` as part of the protos, which we hadn't accounted for. There's also asymmetry in several places because the `TraceFlags` are _not_ serialized into protobufs for the Exemplars (and maybe also for parent context, I forget).

I any case, I had to shim in a bunch of slightly different expected results, which contain the `TraceFlags.getSampled()` which is what we hard-code to on the de-serialization side. I think overall though it probably cleaned up the tests a little and definitely encouraged some reuse.

Reply:
Oh man, I know how painful dealing with these serializations is, especially when it comes to metrics! 😩 thank you for taking the time to update it 🙏